### PR TITLE
Project Form changes.

### PR DESCRIPTION
**Pull request by UlysseFG**

### CODE REVIEW

File: frontend/src/components/projects/ProjectEditContent.tsx
Comment (nafiz1001) [inline frontend/src/components/projects/ProjectEditContent.tsx:1]: Maybe introduce type definitions for the `serialize` and `deserialize` functions that are at the bottom of the code?

File: frontend/src/models/empty_models.ts
Comment (nafiz1001) [inline frontend/src/models/empty_models.ts:1]: You might be able to type those empty models with interfaces in frontend/src/models/frontend_models.ts or frontend/src/models/fms_api_models.ts

Comment (UlysseFG) [reply frontend/src/models/empty_models.ts:1]: I think i'll let this to Chris ... He is the one that pushes for typing and i really don't want to get in his path on that by creating ad hoc types just to make the damn thing stop complaining ...

File: frontend/src/components/containers/ContainerEditContent.js [hunk 1]
```diff
@@ -132,10 +132,13 @@ const ContainerEditContent = ({ token, containerKinds, containersByID, add, upda
...
         >
-          <Item label="Name" {...props("name")} rules={nameRules.concat(requiredRules)}>
+          <Item label="Name" {...props("name")} rules={nameRules.concat(requiredRules)}
+            tooltip="Use [a-z], [A-Z], [0-9], or [ - ][ _ ][ . ]. Space not allowed."
+            extra="Unique name given to a container." >
             <Input />
           </Item>
-          <Item label="Kind" {...props("kind")} rules={requiredRules}>
+          <Item label="Kind" {...props("kind")} rules={requiredRules}
+            extra="Format of the container." >
```
Comment (nafiz1001) [inline frontend/src/components/containers/ContainerEditContent.js:141]: Curious to know why you say "Format" here when it could've been "Kind"

**Review by nafiz1001 — APPROVED** [review]

File: frontend/src/components/containers/ContainerEditContent.js (hunk 1 above)
Comment (UlysseFG) [reply frontend/src/components/containers/ContainerEditContent.js:141]: Defining or explaining a word or an expression using the same expression is not accomplishing much.

**Pull request merged**